app/uk/gov/hmrc/transitmovementsrouter/connectors/PushNotificationsConnector.scala

```diff
@@ -62,34 +67,79 @@ class PushNotificationsConnectorImpl @Inject() (httpClientV2: HttpClientV2, appC
   private def pushNotificationMessageUpdate(movementId: MovementId, messageId: MessageId): UrlPath =
     UrlPath.parse(s"$baseRoute/traders/movements/${movementId.value}/messages/${messageId.value}")
 
-  override def post(movementId: MovementId, messageId: MessageId, source: Source[ByteString, _])(implicit
+  override def post(movementId: MovementId, messageId: MessageId, bodyData: Option[Source[ByteString, _]])(implicit
     hc: HeaderCarrier,
     ec: ExecutionContext
-  ): EitherT[Future, PushNotificationError, Unit] =
-    EitherT {
-      if (!appConfig.pushNotificationsEnabled) Future.successful(Right(()))
-      else {
-        val url = baseUrl.withPath(pushNotificationMessageUpdate(movementId, messageId))
+  ): EitherT[Future, PushNotificationError, Unit] = {
+    val result = if (!appConfig.pushNotificationsEnabled) {
+      Future.successful(Right(()))
+    } else {
+      val url = baseUrl.withPath(pushNotificationMessageUpdate(movementId, messageId))
+
+      val requestBuilder = httpClientV2.post(url"$url").transform(_.addHttpHeaders(HeaderNames.CONTENT_TYPE -> MimeTypes.XML))
```

Comment:
adding the xml content type should be done inside of `case Some(body) =>` since we're not sending any content when the body is a None

Reply:
Okay